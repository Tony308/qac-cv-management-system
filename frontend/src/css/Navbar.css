ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
}
li {
    display: inline;
    float: left;
}
li a {
    display: block;
    padding: 8px;
    background-color: #dddddd;
}

.logoutBtn {
    float: right;
    display: inline;
    background-color: #dddddd;
    padding: 5px;

}