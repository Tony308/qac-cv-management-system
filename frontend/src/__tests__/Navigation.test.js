import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import {render, cleanup} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';

import Navbar from '../containers/NavigationContainer';

afterEach(() => {
    cleanup();
    sessionStorage.removeItem('auth');
})

describe('Navbar ', () => {
    test('snapshot', () => {
        const tree = renderer.create(
            <MemoryRouter>
                <Navbar />
            </MemoryRouter>
            ).toJSON();
        expect(tree).toMatchSnapshot();
    })

    test('smoke test', () => {
        const {getByRole} = render(
            <MemoryRouter>
                <Navbar/>
            </MemoryRouter>
        );
        const home = getByRole('link', {name: 'Home'});
        const createacc = getByRole('link', {name: 'create-account'});

        expect(home).toHaveTextContent('Home');
        expect(createacc).toHaveTextContent('create-account');
    })

    test('logout', () => {
        sessionStorage.setItem('auth', 'true');
        var auth = sessionStorage.getItem('auth');
        expect(auth).toBe('true');

        const {getByRole} = render(
            <MemoryRouter>
                <Navbar/>
            </MemoryRouter>
        )

        const logout = getByRole('button');

        userEvent.click(logout);
        auth = sessionStorage.getItem('auth');
        expect(auth).toBeNull();

    })
})