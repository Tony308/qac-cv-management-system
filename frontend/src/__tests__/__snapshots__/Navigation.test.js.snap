// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar  snapshot 1`] = `
<nav>
  <ul>
    <li>
      <a
        href="/Home"
        onClick={[Function]}
      >
        Home
        <br />
      </a>
    </li>
    <li>
      <a
        href="/create-account"
        onClick={[Function]}
      >
        create-account
        <br />
      </a>
    </li>
     
  </ul>
  <br />
  <br />
  <hr />
</nav>
`;
